File: go/algos/bubbleSort.go
package algos

import "fmt"

func bubbleSort(arr []int) {
	for i := 0; i < len(arr)-1; i++ {
		for j := 0; j < len(arr)-1-i; j++ {
			if arr[j] > arr[j+1] {
				temp := arr[j+1]
				arr[j+1] = arr[j]
				arr[j] = temp
			}
		}
	}
}

func recursiveBubbleSort(arr []int) {
	if len(arr) <= 1 {
		return
	}
	recursiveBubbleSort(arr[1:])
	for i := 0; i < len(arr)-1; i++ {
		if arr[i] > arr[i+1] {
			temp := arr[i+1]
			arr[i+1] = arr[i]
			arr[i] = temp
		}
	}
}

func main() {
	arr := []int{2, 7, 1, 3, 5, 4, 6, 9, 8}
	bubbleSort(arr)
	fmt.Println(arr)
}

File: ts/algos/bubbleSort.ts
// trying bubble sort with js
//
function bubbleSort(arr: number[]): void {
  // arr [ 0 -> n ]
  for (let i = 0; i < arr.length; ++i) {
    for (let j = 0; j < arr.length - 1 - i; ++j) {
      if (arr[j] > arr[j + 1]) {
        const tmp = arr[j];
        arr[j] = arr[j + 1];
        arr[j + 1] = tmp;
      }
    }

    console.log(`Sorted ${i + 1} : ${arr}`);
  }
}

bubbleSort([5, 1, 2, 4, 6, 7, 3])
